src/enum.cr

```diff
@@ -86,6 +86,22 @@
 #   puts "Got blue"
 # end
 # ```
+#
+# ### Changing the Base Type
+#
+# The type of the underlying enum value is `Int32` by default, but it can be changed:
+#
+# ```
+# enum Color : UInt8
+#   Red
+#   Green
+#   Blue
+# end
+#
+# Color::Red.value # => UInt8
+# ```
+#
+# NOTE: Only `Int` based types are allowed.
```

Comment:
The type must be a primitive, not any `Int`. `BigInt` is not possible for example.
```suggestion
# NOTE: Only `Int::Primitive ` types are allowed.
```